rag/utils.py

```diff
@@ -0,0 +1,41 @@
+from sentence_transformers import SentenceTransformer
+import pinecone
+import openai
+import streamlit as st
+
+openaiKey = ""
+pineconeKey = ''
+
+openai.api_key = openaiKey
+model = SentenceTransformer('all-MiniLM-L6-v2')
+
+pinecone.init(api_key=pineconeKey, environment='gcp-starter')
+index = pinecone.Index('demo')
+
+def find_match(input):
+    input_em = model.encode(input).tolist()
```

Comment:
Always a good idea to put comments about what the function does. You can also generate it from ChatGPT -> this is from chatGPT.

    """
    This function takes an input string and encodes it using a pre-trained model. 
    It then queries an index to find the top 3 most similar matches based on the encoded input.
    The function returns the metadata text of these top 3 matches concatenated as a single string,
    separated by newline characters.
    
    Parameters:
    - input (str): The input string to be encoded and matched.
    
    Returns:
    - str: A string containing the metadata text of the top 3 most similar matches, separated by newlines.
    """